drivers/net/wireless/marvell/mwifiex/sta_ioctl.c

```diff
@@ -60,9 +60,10 @@ int mwifiex_wait_queue_complete(struct mwifiex_adapter *adapter,
 	int status;
 
 	/* Wait for completion */
-	status = wait_event_interruptible_timeout(adapter->cmd_wait_q.wait,
-						  *(cmd_queued->condition),
-						  (12 * HZ));
+	status = wait_event_timeout(adapter->cmd_wait_q.wait,
```

Comment:
Does that fix instabilities? If so, is there any other way we can ensure completion/timeout if a user interrupts before sending another command? In general, long uninterruptible waits should be avoided in IOCTLs/on user processes, as this makes the process uninterruptible and also unkillable. At the very least, I think `wait_event_killable_timeout` should be used.

Reply:
I'm not sure if this actually fixes anything, it just gets rid of a bunch of error messages on dmesg about terminated wait queues that happen sometimes. I've removed that commit for now.